### PR TITLE
DDCE-1211: requestBonusPayment error

### PR DESCRIPTION
**Pull request by nathan-rees**
# DDCE-(StoryNumberHere)

## Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/uk/gov/hmrc/lisaapi/LisaModule.scala [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/lisaapi/LisaModule.scala:1]: Cn you split the copyright headers into a separate commit?

File: app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala [hunk 2]
```diff
@@ -286,7 +285,9 @@ class DesConnector @Inject()(
         case SERVICE_UNAVAILABLE => DesUnavailableResponse
         case _ => parseDesResponse[DesTransactionResponse](res)
       }
-    })
+    }).recover{
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala:288]: I'm not sure we want to turn every upstream error response into a DES unavailable? Ideally the recover block should replace the behaviour of the pattern matching on the status.

File: app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala [hunk 3]
```diff
@@ -286,7 +285,9 @@ class DesConnector @Inject()(
...
+      case _: UpstreamErrorResponse => DesUnavailableResponse
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala:289]: Also might be a good idea to have a log here since we're handling an exception.

File: project/plugins.sbt [hunk 4]
```diff
@@ -2,7 +2,7 @@ resolvers += Resolver.url("HMRC Sbt Plugin Releases", url("https://dl.bintray.co
 
 resolvers += "HMRC Releases" at "https://dl.bintray.com/hmrc/releases"
 
-resolvers += "Typesafe Releases" at "http://repo.typesafe.com/typesafe/releases/"
+resolvers += "Typesafe Releases" at "https://repo.typesafe.com/typesafe/releases/"
```
Comment (Ufuoma-Daniel) [inline project/plugins.sbt:6]: Don't forget to update the plugins.

File: app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala [hunk 5]
```diff
@@ -397,4 +401,21 @@ class DesConnector @Inject()(
         }
     }
   }
+  def parseConflictDesResponse(res:UpstreamErrorResponse): DesResponse ={
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala:404]: Hmm some of this code is redundant. I see you're trying to mimic the behaviour of the parseDesResponse function but by taking in an UpstreamErrorResponse, it means a lot of this functionality doesn't apply. e.g. you wouldn't get a 2xx status code as an error response.

File: app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 HM Revenue & Customs
+ * Copyright 2021 HM Revenue & Customs
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/lisaapi/connectors/DesConnector.scala:2]: Minor aside that you don't have to do, but can you update the github checklist template to the smaller one we've been using?

**Review by LetMeTestThat — APPROVED** [review]
